cfme/tests/infrastructure/test_vmware_provider.py

```diff
@@ -66,9 +71,33 @@ def test_appliance_scsi_control_vmware():
         casecomponent: Appliance
         caseimportance: critical
         initialEstimate: 1/4h
-    #TODO: yet to test this, once done, I will add steps. Test was not written by me originally.
     """
-    pass
+    url = ("{baseurl}/{series}/{ver}/cfme-vsphere-paravirtual-{ver}-1.x86_64.vsphere.ova"
+        .format(baseurl=conf.cfme_data['basic_info']['cfme_images_url']['baseurl'],
+            series=appliance.version.series(), ver=appliance.version))
+    logger.info("Downloading ova file for parvirtual vsphere scsi controller test from %s" % url)
+    ova_file = urllib.urlopen(url)
+    filename = os.path.basename(url)
+    with open(filename, 'wb') as output:
+        output.write(ova_file.read())
+
+    @request.addfinalizer
+    def _cleanup():
+        if os.path.exists(filename):
+            os.remove(filename)
+    tar = tarfile.open(filename)
+    desc_member = tar.getmember("desc.ovf")
+    f = tar.extractfile(desc_member)
+    content = f.read()
+    assert content, "No content could be read from desc.ovf"
+    logger.debug("Desc file contains following text:%s" % content)
+    scsi_controller_types = ['lsilogic', 'buslogic', 'lsilogicsas', 'virtualscsi']
+    content = content.lower()
+    counter = 0  # Count any mention of scsi_controller_type in ova XML content
+    for scsi_controller_type in scsi_controller_types:
```

Comment:
Maybe I'm reading this wrong, but I don't think that this is an accurate assertion for the test.

This is checking that three is at least one instance of any of the strings from `scsi_controller_types` in the content - which would yield a false positive for the purpose of the test.

Non-paravirtual images are built with `lsilogic` type SCSI controller, and paravirtual images are built with `virtualscsi` type SCSI controller.  This test needs to tightly assert that `virtualscsi` is the `ResourceSubType` for the `SCSI Controller`.  As I see it now, you could have downloaded the non-paravirtual image and ran this test, and it would pass, because the `lsilogic` controller would have triggered this count increasing.

Did you consider using `xml.etree` to actually parse the OVF file?

I think the minimum assertion here should be that there is one (at least) `ResourceSubType` element in the tree with a value of 'virtualscsi'.